mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,563 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+MULTI_CLASSIFIER_NAME = 'ssim_s2_cloud_detector_lightGBM_v0.2.joblib.dat'
+
+class AddMultiCloudMaskTask(EOTask):
+    """ This task wraps around s2cloudless and the SSIM-based multi-temporal classifier.
+    Its intended output is a cloud mask that is based on the outputs of both
+    individual classifiers (a dilated intersection of individual binary masks).
+    Additional cloud masks and probabilities can be added for either classifier or both.
+
+    Prior to feature extraction and classification, it is recommended that the input be
+    downscaled by specifying the source and processing resolutions. This should be done
+    for the following reasons:
+        - faster execution
+        - lower memory consumption
+        - noise mitigation
+
+    Resizing is performed with linear interpolation. After classification, the cloud
+    probabilities are themselves upscaled to the original dimensions, before proceeding
+    with masking operations.
+    """
+
+    def __init__(self,
+                 mono_classifier=None,
+                 multi_classifier=None,
+                 data_feature='BANDS-S2-L1C',
+                 is_data_feature='IS_DATA',
+                 all_bands=True,
+                 processing_resolution=None,
+                 max_proc_frames=11,
+                 mono_features=None,
+                 multi_features=None,
+                 mask_feature='CLM_INTERSSIM',
+                 mono_threshold=0.4,
+                 multi_threshold=0.5,
+                 average_over=1,
+                 dilation_size=1
+                ):
+        """Constructor.
+
+        :param mono_classifier: Classifier used for mono-temporal cloud detection (`s2cloudless` or equivalent).
+                                Default value: None (s2cloudless is used)
+        :type mono_classifier: sklearn Estimator that works on the 10 selected reflectance bands
+                                (`B01`, `B02`, `B04`, `B05`, `B08`, `B8A`, `B09`, `B10`, `B11`, `B12`)
+        :param multi_classifier: Classifier used for multi-temporal cloud detection.
+                                 Default value: None (SSIM-based model is used)
+        :type multi_classifier: sklearn Estimator that works on the 90 multi-temporal features:
+                                 - raw reflectance value in the target frame,
+                                 - average value within a spatial window in the target frame,
+                                 - maximum, mean and standard deviation of the structural similarity (SSIM)
+                                 - indices between a spatial window in the target frame and every other,
+                                 - minimum and mean reflectance of all available time frames,
+                                 - maximum and mean difference in reflectances between the target frame and every other.
+        :param data_feature: Name of the key in the `eopatch.data` dictionary, which stores raw reflectance data.
```

Comment:
new tasks use tuples of `FeatureType` and `str` to define a feature and parse them with `self._parse_features`. Might be good to apply this where possible, although can be changed later on. The allowed `FeatureType` can be added to the parser (e.g. `Featuretype.DATA` for bands and `FeatureType.MASK` for `is_data`)

Reply:
If I understand correctly `self._parse_features` is intended for input features. Is this also needed for output features, or can they remain just strings (since the type of the feature is determined in the task)?

Reply:
it's usually used for input features, unless the output feature type needs to be specified by the user

Reply:
Okay great. The input features are now parsed using `self._parse_features`. I've left the output features as they are.